pkg/controller/mongodb/mongodb_tls.go

```diff
@@ -0,0 +1,144 @@
+package mongodb
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/mongodb/mongodb-kubernetes-operator/pkg/kube/container"
+	"github.com/mongodb/mongodb-kubernetes-operator/pkg/kube/podtemplatespec"
+	"github.com/mongodb/mongodb-kubernetes-operator/pkg/kube/statefulset"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+
+	mdbv1 "github.com/mongodb/mongodb-kubernetes-operator/pkg/apis/mongodb/v1"
+)
+
+type tlsValidationResult struct {
+	Valid   bool
+	Message string
```

Comment:
can we make this unexported? I don't think we should have an unexported type with exported fields

Reply:
Yes, should definitely be unexported